stripecardscan/src/main/java/com/stripe/android/stripecardscan/cardscan/CardScanFragment.kt

```diff
@@ -0,0 +1,306 @@
+package com.stripe.android.stripecardscan.cardscan
+
+import android.annotation.SuppressLint
+import android.content.res.Resources
+import android.graphics.Bitmap
+import android.graphics.PointF
+import android.os.Bundle
+import android.util.Size
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.core.os.bundleOf
+import androidx.core.view.updateMargins
+import androidx.fragment.app.setFragmentResult
+import com.stripe.android.camera.CameraPreviewImage
+import com.stripe.android.camera.framework.Stats
+import com.stripe.android.stripecardscan.R
+import com.stripe.android.stripecardscan.cardscan.exception.InvalidStripePublishableKeyException
+import com.stripe.android.stripecardscan.cardscan.exception.UnknownScanException
+import com.stripe.android.stripecardscan.cardscan.result.MainLoopAggregator
+import com.stripe.android.stripecardscan.cardscan.result.MainLoopState
+import com.stripe.android.stripecardscan.databinding.FragmentCardscanBinding
+import com.stripe.android.stripecardscan.framework.api.dto.ScanStatistics
+import com.stripe.android.stripecardscan.framework.api.uploadScanStats
+import com.stripe.android.stripecardscan.framework.util.AppDetails
+import com.stripe.android.stripecardscan.framework.util.Device
+import com.stripe.android.stripecardscan.payment.card.ScannedCard
+import com.stripe.android.stripecardscan.scanui.CancellationReason
+import com.stripe.android.stripecardscan.scanui.ScanErrorListener
+import com.stripe.android.stripecardscan.scanui.ScanFragment
+import com.stripe.android.stripecardscan.scanui.ScanState
+import com.stripe.android.stripecardscan.scanui.SimpleScanStateful
+import com.stripe.android.camera.scanui.util.asRect
+import com.stripe.android.stripecardscan.scanui.util.getColorByRes
+import com.stripe.android.stripecardscan.scanui.util.getFloatResource
+import com.stripe.android.stripecardscan.scanui.util.startAnimation
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.launch
+import java.util.concurrent.atomic.AtomicBoolean
+import kotlin.math.min
+import kotlin.math.roundToInt
+
+private val MINIMUM_RESOLUTION = Size(1067, 600) // minimum size of OCR
+const val CARD_SCAN_FRAGMENT_REQUEST_KEY = "CardScanRequestKey"
+const val CARD_SCAN_FRAGMENT_BUNDLE_KEY = "CardScanBundleKey"
+const val CARD_SCAN_FRAGMENT_PARAMS_KEY = "CardScanParamsKey"
+
+class CardScanFragment : ScanFragment(), SimpleScanStateful<CardScanState> {
+
+    override val minimumAnalysisResolution = MINIMUM_RESOLUTION
+
+    private lateinit var viewBinding: FragmentCardscanBinding
+
+    override val instructionsText: TextView by lazy { viewBinding.instructions }
+
+    override val previewFrame: ViewGroup by lazy { viewBinding.previewFrame }
+
+    private val params: CardScanSheetParams by lazy {
+        CardScanSheetParams(arguments?.getString(CARD_SCAN_FRAGMENT_PARAMS_KEY) ?: "")
```

Comment:
Because `CardScanSheetParams` is `internal`, it's not currently possible to launch the fragment with a valid CardScanSheetParams from an external module.

Can you create a method in the `CardScanSheet` class for handling the fragment? It would look something like this:

```
    fun attachCardScanFragmentArgs(
        supportFragmentManager: SupportFragmentManager,
        @IdRes fragmentContainer: Int,
        onFinished: (cardScanSheetResult: CardScanSheetResult) -> Unit,
    ) {
        supportFragmentManager.commit {
            setReorderingAllowed(true)
            add<CardScanFragment>(fragmentContainer, args = bundleOf(CARD_SCAN_FRAGMENT_PARAMS_KEY to CardScanSheetParams(stripePublishableKey)))
        }

        supportFragmentManager
            .setFragmentResultListener(CARD_SCAN_FRAGMENT_REQUEST_KEY, this) { _, bundle ->
                onFinished(bundle.getParcelable(CARD_SCAN_FRAGMENT_BUNDLE_KEY))
            }
    }
```